primitives/src/authorization.rs

```diff
@@ -0,0 +1,41 @@
+use crate::identity_id::IdentityId;
+use codec::{Decode, Encode};
+use rstd::prelude::Vec;
+
+/// Authorization data for two step prcoesses.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum AuthorizationData {
+    /// Authorization to transfer a ticker
+    TransferTicker(Vec<u8>),
+    /// Any other authorization
+    Custom(Vec<u8>),
+    /// No authorization data
+    None,
+}
+
+impl Default for AuthorizationData {
+    fn default() -> Self {
+        AuthorizationData::None
+    }
+}
+
+/// Authorization struct
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug)]
+pub struct Authorization<U> {
+    /// Enum that contains authorization type and data
+    pub authorization_data: AuthorizationData,
+
+    /// Identity of the organization/individual that added this authorization
+    pub authorized_by: IdentityId,
+
+    /// time when this authorization expires. optional.
+    pub expiry: Option<U>,
+
+    // Extra data to allow iterating over the authorizations.
+    /// Authorization number of the next Authorization
```

Comment:
Doc comments will compile into a single line without a line break, so a `.` would look better at the end of sentences.

Reply:
Makes sense, I'll add them.